pandas/core/indexes/datetimes.py

```diff
@@ -588,7 +588,9 @@ def _generate(cls, start, end, periods, name, freq,
             index = index[1:]
         if not right_closed and len(index) and index[-1] == end:
             index = index[:-1]
-        index = cls._simple_new(index, name=name, freq=freq, tz=tz)
+
+        index = cls._simple_new(index.values, name=name, freq=freq, tz=tz)
```

Comment:
i don’t like having 2 different construction paths generally: eg we always need to be an ndarray or already converted to a DTI by the time _simple_new gets called

what i would do is run all of the index tests and see what the current state is
then probably settle on an ndarray input to _simple_new and put an assertion to validate this

Reply:
Which 2 different construction paths do you mean? 
With the update above, `index` is always an index, and it's always the values that are passed to `_simple_new`. So it is ndarray input to `_simple_new`.

Reply:
my point is that we need an assertion to validate this. it may be that everything is fixed, but we should actually test this.